mnemosyne/diagnose.py

```
@@ -62,6 +62,118 @@ def _safe_env(name: str) -> str:
+                     WHERE NOT EXISTS (
+                         SELECT 1 FROM live_ids WHERE live_ids.id = memory_embeddings.memory_id
+                     )
+                 )
+            SELECT
+                (SELECT COUNT(*) FROM gists) AS gists_total,
+                (SELECT COUNT(*) FROM gists WHERE memory_id IS NOT NULL) AS gists_with_memory_id,
+                (SELECT COUNT(*) FROM orphan_gists) AS gists_orphan_memory_id,
+                (SELECT COUNT(*) FROM memory_embeddings) AS memory_embeddings_total,
+                (SELECT COUNT(*) FROM orphan_embeddings) AS memory_embeddings_orphan_memory_id,
+                (SELECT COUNT(*) FROM orphan_gists INNER JOIN orphan_embeddings USING (memory_id)) AS orphan_memory_id_overlap
+            """
```

Ruff (0.15.20)

[error] Possible SQL injection vector through string-based query construction

(S608)